From 78d1b66ad3ba0f2ee7296be37cf0c1dd9a360b34 Mon Sep 17 00:00:00 2001
From: Daniel Petry <dpetry@anaconda.com>
Date: Tue, 21 Jan 2025 17:45:23 -0600
Subject: [PATCH 09/16] Add conda prefix to inductor include paths

Currently inductor doesn't look in conda's includes and libs. This results in
errors when it tries to compile, if system versions are being used of
dependencies (e.g., sleef).

Note that this is for inductor's JIT mode, not its AOT mode, for which the
end user provides a <filename>_compile_flags.json file.
---
 torch/_inductor/cpp_builder.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/torch/_inductor/cpp_builder.py b/torch/_inductor/cpp_builder.py
index d0a646c1de8..7468f2787d9 100644
--- a/torch/_inductor/cpp_builder.py
+++ b/torch/_inductor/cpp_builder.py
@@ -1136,6 +1136,7 @@ def get_cpp_torch_options(
         + python_include_dirs
         + torch_include_dirs
         + omp_include_dir_paths
+        + [os.getenv('CONDA_PREFIX') + '/include']
     )
     cflags = sys_libs_cflags + omp_cflags
     ldflags = omp_ldflags
